packages/micromark-util-subtokenize/dev/index.js

```diff
@@ -246,10 +246,11 @@ function subcontent(events, eventIndex) {
     const slice = childEvents.slice(breaks[index], breaks[index + 1])
     const start = startPositions.pop()
     assert(start !== undefined, 'expected a start position when splicing')
-    jumps.unshift([start, start + slice.length - 1])
+    jumps.push([start, start + slice.length - 1])
```

Comment:
It looks like jumps is an array of `[number, number]` tuples, which always ends up having the same length as `breaks.length`.

If this is such a hot path, maybe we can optimize it further by:

- Making `jumps` an array of fixed length instead of pushing.
- Splitting `jumps` into 2 arrays of `number[]`.
- Turning those two arrays into arrays of type `UintNArray`, where I don’t know the appropriate value of `N`.

Reply:
Yes! But I’d personally really want to look at the edit map data structure which I used in Rust: https://github.com/wooorm/markdown-rs/blob/60db8e5896be05d23137a6bdb806e63519171f9e/src/util/edit_map.rs#L1-L9.